site/src/pages/components/customizableselect.mdx

```diff
@@ -286,11 +275,9 @@ Here is another example with custom content in the listbox: [codepen](https://co
 <button><selectedoption id=myselectedoption></selectedoption></button>
 <select selectedoptionelement=myselectedoption>
```

Comment:
i know everyone really wants to use this for split buttons, but we should come to consensus on what a good markup pattern would be for that, and if there are any additional accessibility considerations we should make before we actively promote it as an example.  Or, in lieu of having a more robust pattern - we should explicitly note that this is merely a simplified version of only the _start_ of what someone would need to do to create a split button, and this markup pattern is not a recommendation in and of itself.

Reply:
cc @smhigley

Reply:
I don't like the idea of saying that this new attribute we're adding to the platform is so hard to use that we can't show you how to use it correctly.

The text inside the selectedoption inside the button accurately describes what action the button will do when activated. What would be better?

Reply:
well, fortunately i'm not saying it's going to be too hard to use.  but i am saying we haven't really discussed what we should promote as a markup pattern (or patterns) for this - so we should do that.

for instance, it seems you're assuming i have a problem with the button the selectedoption is placed within - but that's not my concern.  Rather, what is one to expect the select's accessible name should be per the pattern you wrote?  Nothing?  Down arrow? neither of those are particularly good answers.  

so how do we want to recommend people label the select?  or should we consider ways to automatically name the select / associate it somehow with the selectedoption still?  

also, commonly component libraries have coded the 'more options' portion of the split button (the select) as a button that invokes a role=menu popup.  My assumption with what we were doing here is simply giving people a way to more easily markup their custom split buttons, and then it would be on authors to finish the rest of the setup on their own.  but am i wrong in assuming we were on the same page?  is there appetite in implementing more of this for authors automatically - to the point that your simple markup pattern _would_ do more by default, and thus would then be sufficient?

Reply:
another question.  would the button part be necessary for this sort of pattern, or if one were to leave it out, would the existence of the `selectedoptionelement` attribute on the `select` be enough for the browser to know this is a customized select, and render a UA default down chevron for the select, without someone having to add in their own custom button part/icon?

Reply:
It's been a while since I've looked at this, but I'm going to jump in since I've done some work on both splitbuttons and custom combos & their accessibility.

I agree with @scottaohara, this pattern has a number of accessibility implications that are not addressed or handled explicitly. Splitbuttons have a history as a Windows native control that combines a primary action with a menu, but does not easily map to the web for reasons including the following:
- Having two actions for one control causes problems for different access modalities, especially outside of a desktop environment (e.g. voice and touch screen readers)
- Splitting it out into two separate control types helps with this, but raises questions around tab/arrow focus targets and behavior as well as naming (as @scottaohara already brought up).
- Roles are a little difficult. There's a control type for this in Windows, but not in other platforms, at least not that I'm aware of. Splitting this into two controls means it's even more unclear how you would create markup that shows the two buttons(?) are related.

Additionally, splitbuttons generally open a menu, not a listbox, and a `<select>` is not a good fit for the menu portion. Since `<select>` is a form field, it has certain screen reader navigation & shortcut behaviors that do not make sense for the menubutton portion of a splitbutton. It also exposes a value, and it's not clear how that would be presented to the user in an understandable way when that value is presented to the user in the primary action button but not in the combobox/menu button.

I get the motivation behind making a splitbutton, but I think any effort to do so would be better served by being independent of `<select>` work. Trying to duct tape them together seems like it would degrade the accessibility and usability of both the `<select>` and the desired splitbutton pattern.

Reply:
Also stepping back a bit from the splitbutton question -- I think I might be misunderstanding this because I've been out of the loop, but is `selectedoption` essentially just a way for a developer to drop the text of a `<select>`'s selected value anywhere on the page, or is it more constrained?

The idea of having an HTML mechanism to reflect the current value of a form control seems like quite a cool idea, would it make sense to have that mechanism be more general than just `<select>`? I could see essentially the same use case for being able to reflect the value of radios, sliders, general text-like inputs, etc.